src/DotNetCore.CAP.Redis/IRedisStream.Manager.Default.cs

```diff
@@ -0,0 +1,122 @@
+            }
+            else
+            {
+                var groupInfo = await database.StreamGroupInfoAsync(stream);
+                if (groupInfo.Any(g => g.Name == consumerGroup))
+                    return;
+                await database.StreamCreateConsumerGroupAsync(stream, consumerGroup, StreamPosition.NewMessages);
+            }
+        }
+
+        public async Task PublishAsync(string stream, NameValueEntry[] message)
+        {
+            await ConnectAsync();
+
+            //The object returned from GetDatabase is a cheap pass - thru object, and does not need to be stored
+            await redis.GetDatabase().StreamAddAsync(stream, message);
+        }
+
+        public async IAsyncEnumerable<RedisStream[]> PollStreamsLatestMessagesAsync(string[] streams, string consumerGroup, TimeSpan pollDelay, [EnumeratorCancellation] CancellationToken token)
+        {
+            var positions = streams.Select(stream => new StreamPosition(stream, StreamPosition.NewMessages));
+
+            while (true)
+            {
+                var result = await TryReadConsumerGroup(consumerGroup, positions.ToArray(), token).ConfigureAwait(false);
+
+                yield return result.streams;
+
+                token.WaitHandle.WaitOne(pollDelay);
+            }
+        }
+
+        public async IAsyncEnumerable<RedisStream[]> PollStreamsPendingMessagesAsync(string[] streams, string consumerGroup, TimeSpan pollDelay, [EnumeratorCancellation] CancellationToken token)
+        {
+            var positions = streams.Select(stream => new StreamPosition(stream, StreamPosition.Beginning));
+
+            while (true)
+            {
+                token.ThrowIfCancellationRequested();
+
+                var result = await TryReadConsumerGroup(consumerGroup, positions.ToArray(), token).ConfigureAwait(false);
+
+                yield return result.streams;
+
+                //Once we consumed our history of pending messages, we can break the loop.
+                if (result.canRead && result.streams.All(s => s.Entries.Length < options.StreamEntriesCount))
+                    break;
+
+                token.WaitHandle.WaitOne(pollDelay);
+            }
+        }
+
+        private async Task<(bool canRead, RedisStream[] streams)> TryReadConsumerGroup(string consumerGroup, StreamPosition[] positions, CancellationToken token)
+        {
+            try
+            {
+                await ConnectAsync();
+
+                var database = redis.GetDatabase();
+
+                var readSet = database.StreamReadGroupAsync(positions, consumerGroup, consumerGroup, (int)options.StreamEntriesCount);
+
+                return (true, await readSet.ConfigureAwait(false));
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Redis error when trying read consumer group {consumerGroup}", ex);
+                return (false, Array.Empty<RedisStream>());
+            }
```

Comment:
According to my test, we will encounter two exception here.

1. When the Redis server restart, the first exception will be encountered

> SocketFailure (ReadSocketError/ConnectionReset, last-recv: 5) on 192.168.3.57:6379/Interactive, Idle/Faulted, last: 
XREADGROUP

2. After the redis server restarts, the consumer group will be lost, The exception information is as follows

> NOGROUP No such key 'test-message' or consumer group 'Samples.Redis.SqlServer.v1' in XREADGROUP with GROUP option

Reply:
PS: If transport throw `BrokerConnectionException` exception,  the `TransportCheckProcessor`  will restart consumer every 30  seconds